admin/server/deployment.go

```diff
@@ -124,7 +124,7 @@ func (s *Server) GetDeploymentCredentials(ctx context.Context, req *adminv1.GetD
 	permissions := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID)
 
 	// If the user is not a superuser, they must have ManageProd permissions
-	if !permissions.ManageProd && !claims.Superuser(ctx) {
```

Comment:
we are disallowing deploy access to superusers now, if yes then comment above should be updated